eng/pipelines/templates/jobs/ci.tests.yml

```
@@ -63,17 +76,26 @@ jobs:
       - template: /eng/pipelines/templates/steps/install-dotnet.yml
         parameters:
           NuGetCacheKey: Test
+      - pwsh: | 
+          if ("${{ parameters.ProjectListOverrideFilePropertyName }}" -ne '') {
+            Write-Host "##vso[task.setvariable variable=ProjectListFileName]$(${{ parameters.ProjectListOverrideFilePropertyName }})"
+          }
+          else 
+          {
+            Write-Host "##vso[task.setvariable variable=ProjectListFileName]"
+          }
+        displayName: Set variable for the project list file 
       - script: >-
           dotnet test eng/service.proj --filter "(TestCategory!=Manually) & (TestCategory!=Live)" --framework $(TestTargetFramework)
           --logger "trx;LogFileName=$(TestTargetFramework).trx" --logger:"console;verbosity=normal"
           --blame-crash-dump-type full --blame-hang-dump-type full --blame-hang-timeout ${{parameters.TestTimeoutInMinutes}}minutes
           /p:SDKType=${{ parameters.SDKType }}
-          /p:ServiceDirectory=${{ parameters.ServiceToTest }}
+          /p:ServiceDirectory=${{ parameters.ServiceDirectory }}
           /p:IncludeSrc=false /p:IncludeSamples=false /p:IncludePerf=false /p:IncludeStress=false
           /p:RunApiCompat=false /p:InheritDocEnabled=false
           /p:Configuration=$(BuildConfiguration)
           /p:CollectCoverage=$(CollectCoverage) /p:CodeCoverageDirectory=$(Build.SourcesDirectory)\sdk\${{parameters.ServiceDirectory}}
-          /p:EnableSourceLink=false
+          /p:EnableSourceLink=false /p:ProjectListOverrideFile=$(ProjectListFileName)
```

```suggestion
          /p:EnableSourceLink=false 
          /p:ProjectListOverrideFile=$env:${{ parameter.ProjectListOverrideFilePropertyName }}
```

doing it this way should make it empty if the property isn't set but will pass in the value if it is set.

---

Did this approach not work for some reason?

---

I see why this doesn't work if this property name is empty.